### DISCUSSION TITLE
Export the ShouldBlockFnArgs type from useBlocker

### DISCUSSION THREAD

**Discussion by erictompkins** [TanStack/router#3030] [Ideas]
The merged #1790 pull request is great and has helped us out. The `args` parameter on the `ShouldBlockFn` function was exactly what we needed. 

Can you also export the `ShouldBlockFnArgs` type for the function `arg` parameter so that we can use it if we set up a function call?

For example, we're doing something like this:

```jsx
<Block shouldBlockFn={(args) => getUseBlocker(args)} withResolver>
    {/* CUSTOM UI HERE */}
</Block>
```

We have this function:

```js
const getUseBlocker(args): boolean => {
    const returnValue = false;
    // Tests to determine if to block
    return returnValue;
}
```

We'd like to do this:

```js
const getUseBlocker(args: ShouldBlockFnArgs): boolean => {
}
```

but can't because `ShouldBlockFnArgs` is not exported.

Is it possible to export that type?

Thank you!

**Comment by schiller-manuel** [comment]
can you create a PR for this? should be exportable
